raid-ddf.ks.in

```diff
@@ -0,0 +1,49 @@
+#version=DEVEL
+#test name: raid-ddf
+# This test covers https://bugzilla.redhat.com/show_bug.cgi?id=2063791
+
+%ksappend repos/default.ks
+%ksappend common/common_no_storage_and_payload.ks
+%ksappend payload/default_packages.ks
+# Validate on the first boot.
+%ksappend validation/success_on_first_boot.ks
+
+clearpart --all --initlabel
+reqpart --add-boot
+part swap --size=512
+part / --size=1024 --grow --label=rootfs
+
+%pre
+# Create DDF RAID device before Anaconda scans the present disks
+mdadm --create /dev/md/ddf --name=ddf --level container --raid-disks=2 --metadata=ddf /dev/vda /dev/vdb
+mdadm --create /dev/md/test-raid-ddf --name=test-raid-ddf --level raid1 --raid-disks=2 /dev/md/ddf
+%end
+
+%post
+# Check whether the RAID device was used for installation as expected
+devlink=$(readlink /dev/md/test-raid-ddf)
+if ! lsblk -nl -o MOUNTPOINT /dev/md/${devlink} | grep "^/$"; then
+    echo "*** DDF RAID device /dev/md/test-raid-ddf wasn't used for installation" > /root/RESULT
+    echo -e "\n\nlsblk output:\n" >> /root/RESULT
+    lsblk -i >> /root/RESULT
+    # Power off the VM, so that the error message is retained in /root/RESULT
+    # (otherwise it would get overwritten after a successful boot)
+    poweroff
+fi
+
+cat <<EOF > /usr/libexec/kickstart-test.sh
+# Check that the system booted with the RAID device assembled
+devlink=$(readlink /dev/md/test-raid-ddf)
+if ! lsblk -nl -o MOUNTPOINT /dev/md/${devlink} | grep -q "^/$"; then
+    echo "*** The system hasn't booted with the DDF RAID device /dev/md/test-raid-ddf assembled." > /root/RESULT
+    echo -e "\n\nlsblk output:\n" >> /root/RESULT
+    lsblk -i >> /root/RESULT
+    # Power off the VM, so that the error message is retained in /root/RESULT
+    # (otherwise it would get overwritten after a successful boot)
+    poweroff
```

Comment:
I think the poweroff is not needed here as it is actually the successful boot now.